nupic/research/frameworks/pytorch/imagenet/imagenet_experiment.py

```diff
@@ -395,21 +408,30 @@ def create_lr_scheduler(cls, config, optimizer, total_batches):
 
     @classmethod
     def create_loaders(cls, config):
-        """Create train, val, and test dataloaders."""
+        """Create train and val dataloaders."""
+
+        dataset_class = config.get("dataset_class", None)
+        if dataset_class is None:
+            raise ValueError("Must specify 'dataset_class' in config.")
 
-        create_datasets = config.get("create_datasets_func", None)
-        if create_datasets is None:
-            raise ValueError("Must specify 'create_datasets_func' in config.")
         dataset_args = config.get("dataset_args", {})
-        train_set, val_set, test_set = create_datasets(**dataset_args)
+        dataset_args.update(train=True)
+        train_set = dataset_class(**dataset_args)
+        dataset_args.update(train=False)
+        val_set = dataset_class(**dataset_args)
 
         train_loader = cls.create_train_dataloader(train_set, config)
         val_loader = cls.create_validation_dataloader(val_set, config)
-        test_loader = None
-        if test_set is not None:
-            test_loader = cls.create_validation_dataloader(test_set, config)
 
-        return train_loader, val_loader, test_loader
+        return train_loader, val_loader
+
+    @classmethod
+    def create_train_sampler(cls, dataset, config):
+        if config.get("distributed", False):
+            sampler = DistributedSampler(dataset)
+        else:
+            sampler = None
+        return sampler
 
     @classmethod
     def create_train_dataloader(cls, dataset, config):
```

Comment:
Now that you've changed this to use simpler dataset factory functions, I think it's worth changing these methods back to the previous signatures, `create_train_dataloader(cls, config)` and `create_validation_dataloader(cls, config)` (without the `dataset` input) so that they're easier to use in test  scripts, quantization scripts, etc. We don't need the new `create_loaders` method any more. These methods can create their own datasets easily.

This is only tangentially related to this PR, so feel free to do it in a later PR. (If you agree.)

Reply:
I don't have a strong opinion about this, I would be fine either way. Maybe @mvacaporale can chip in. You are right that as we refactored the need for a `create_loaders` function may no longer be there, and I see your point on having a function that doesn't take dataset as argument as being easier to use. I honestly don't remember why we decided to go with that approach it in the first place.

Reply:
I think the motivation was for the `create_loaders`. We had one dataset function that made both the train and val sets, and so we needed one function to create the loaders. But this is not needed anymore as you both pointed out.

I can make the change after this PR is merged.